### PR TITLE
CMake cleanups

### PR DESCRIPTION
**Pull request by SoapGentoo**
We have no consumers of `WANT_CMAKE` left in tree and removing it makes the eclass simpler.

### CODE REVIEW

**Comment by a17r** [comment]
lgtm, but I don't usually deal with cmake-utils.eclass

**Comment by mgorny** [comment]
Didn't you want to kill EAPI=2 as well?

**Comment by SoapGentoo** [comment]
@mgorny sure, but that (actually EAPI 0-4) has one outstanding stablereq. Once it's done, I will kill that too.

File: eclass/cmake-utils.eclass [hunk 1]
```diff
@@ -588,7 +586,7 @@ enable_cmake-utils_src_configure() {
 	[[ "${NOCOLOR}" = true || "${NOCOLOR}" = yes ]] && echo 'SET (CMAKE_COLOR_MAKEFILE OFF CACHE BOOL "pretty colors during make" FORCE)' >> "${common_config}"
 
 	# Wipe the default optimization flags out of CMake
-	if [[ ${CMAKE_BUILD_TYPE} != Gentoo ]] && ! has "${EAPI}" 2 3 4 5; then
+	if [[ ${CMAKE_BUILD_TYPE} != Gentoo ]] && ! has "${EAPI}" 5; then
```
Comment (mgorny) [inline eclass/cmake-utils.eclass]: Maybe use `[[ ... ]]` matching, `has` really looks weird with a single value ;-P.

File: eclass/cmake-utils.eclass [hunk 2]
```diff
@@ -112,6 +105,7 @@ inherit toolchain-funcs multilib ninja-utils flag-o-matic eutils \
 EXPORT_FUNCTIONS src_prepare src_configure src_compile src_test src_install
 
 [[ ${WANT_CMAKE} ]] && eqawarn "\${WANT_CMAKE} has been removed and is a no-op now"
+[[ ${PREFIX} ]] && eqawarn "\${PREFIX} has been removed and is a no-op now"
```
Comment (mgorny) [inline eclass/cmake-utils.eclass]: I'd say it's a potentially breaking change, so maybe `die` instead.

**Pull request closed without merging**